### PR TITLE
make StackingSensor public

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)
It's not currently possible for users to use our stacking with custom ISensor implementations. This makes StackingSensor public so that they can use it for that.

### Useful links (Github issues, JIRA tickets, ML-Agents forum threads etc.)

https://github.com/Unity-Technologies/ml-agents/issues/3676
https://jira.unity3d.com/browse/MLA-792

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: com.unity.ml-agents/Runtime/Sensors/StackingSensor.cs [hunk 1]
```diff
@@ -6,6 +6,8 @@ namespace MLAgents.Sensors
     /// For example, 4 stacked sets of observations would be output like
     ///   |  t = now - 3  |  t = now -3  |  t = now - 2  |  t = now  |
     /// Internally, a circular buffer of arrays is used. The m_CurrentIndex represents the most recent observation.
+    ///
+    /// Currently, compressed and multidimensional observations are not supported.
     /// </summary>
     internal class StackingSensor : ISensor
```
Comment (surfnerd) [inline com.unity.ml-agents/Runtime/Sensors/StackingSensor.cs]: was this intended to be public?

Comment (chriselion) [reply com.unity.ml-agents/Runtime/Sensors/StackingSensor.cs]: ugh, forgot to do the main point of the PR.

Comment (chriselion) [reply com.unity.ml-agents/Runtime/Sensors/StackingSensor.cs]: Fixed. Took me a while to remove the palm from my face.

File: com.unity.ml-agents/Tests/Editor/PublicAPI/PublicApiValidation.cs [hunk 2]
```diff
@@ -83,6 +83,30 @@ public override float[] Heuristic()
             }
         }
 
+        // Simple SensorComponent that sets up a StackingSensor
+        class StackingComponent : SensorComponent
+        {
+            public SensorComponent wrappedComponent;
```
Comment (vincentpierre) [inline com.unity.ml-agents/Tests/Editor/PublicAPI/PublicApiValidation.cs]: Where is this SensorComponent living? If it is attached to the same agent, won't we have duplicate sensor data?

Comment (chriselion) [reply com.unity.ml-agents/Tests/Editor/PublicAPI/PublicApiValidation.cs]: I commented below that this isn't meant to be practical, it's just an example for the purposes of making sure we have the right access in this test.

https://github.com/Unity-Technologies/ml-agents/blob/57548acfd0cb41e3ced1ff98713741c250f75487/com.unity.ml-agents/Tests/Editor/PublicAPI/PublicApiValidation.cs#L142

**Comment by chriselion** [comment]
not waiting for latest yamato package tests

**Pull request merged**